addons/web/static/tests/core/owl_dialog_tests.js

```diff
@@ -0,0 +1,216 @@
+
+        await testUtils.dom.click(document.querySelector('div.modal button.btn-primary'));
+
+        assert.containsNone(document.body, '.modal',
+            "Modal should be closed");
+
+        owlDialog.close();
+    });
+
+    QUnit.test("Closing dialog with custom buttons by clicking on buttons", async function(assert) {
+        assert.expect(3);
+
+        const testPromise = testUtils.makeTestPromiseWithAssert(assert, 'custom callback');
+
+        const owlDialog = await createOwlDialog({
+            env: {
+                _t: str => str
+            },
+            props: {
+                buttons: [
+                    {
+                        text: "Close",
+                        classes: 'btn-primary',
+                        close: true,
+                        click: testPromise.resolve,
+                    },
+                ],
+                onForceClose: testPromise.reject,
+            }
+        });
+
+        assert.verifySteps([]);
+
+        await testUtils.dom.click(document.querySelector('div.modal button.btn-primary'));
+        await testPromise;
+
+        assert.verifySteps(['ok custom callback']);
+
+        owlDialog.close();
+    });
+
+    QUnit.test("Closing dialog without clicking on buttons", async function (assert) {
+        assert.expect(3);
+
+        const testPromise = testUtils.makeTestPromiseWithAssert(assert, 'custom callback');
+        const owlDialog = await createOwlDialog({
+            env: {
+                _t: str => str
+            },
+            props: {
+                onForceClose: testPromise.resolve,
+            }
+        });
+
+        assert.verifySteps([]);
+
+        await testUtils.dom.triggerEvent(document.querySelector('div.modal[role="dialog"]'), 'keydown', { which: 27 });
+        await testPromise;
+
+        assert.verifySteps(['ok custom callback']);
+
+        owlDialog.close();
+    });
+
+    QUnit.test("Closing dialog with custom buttons without clicking on buttons", async function (assert) {
+        assert.expect(3);
+
+        const testPromise = testUtils.makeTestPromiseWithAssert(assert, 'custom callback');
+        const owlDialog = await createOwlDialog({
+            env: {
+                _t: str => str
+            },
+            props: {
+                buttons: [
+                    {
+                        text: "Close",
+                        classes: 'btn-primary',
+                        close: true,
+                        click: testPromise.reject,
+                    },
+                ],
+                onForceClose: testPromise.resolve,
+            }
+        });
+
+        assert.verifySteps([]);
+
+        await testUtils.dom.triggerEvent(document.querySelector('div.modal[role="dialog"]'), 'keydown', { which: 27 });
+        await testPromise;
+
+        assert.verifySteps(['ok custom callback']);
+
+        owlDialog.close();
+    });
+
+    QUnit.skip("Reactivity to props update", async function (assert) {
```

Comment:
let  us not forget to rewrite this. the idea was just to test reactivity to updates (button disabled or other things like that)

Reply:
ok